near/omni-bridge/src/lib.rs

```diff
@@ -1405,6 +1483,27 @@ impl Contract {
 }
 
 impl Contract {
+    // Hashing the transfer message to calculate virtual account ID that can be used to deposit storage required for the message
+    fn calculate_message_storage_account_id(
+        message: &TransferMessage,
+        signer_id: &AccountId,
+    ) -> AccountId {
+        let mut data = Vec::new();
+        data.extend_from_slice(message.token.to_string().as_bytes());
+        data.extend_from_slice(&message.amount.0.to_le_bytes());
+        data.extend_from_slice(message.recipient.to_string().as_bytes());
+        data.extend_from_slice(&message.fee.fee.0.to_le_bytes());
+        data.extend_from_slice(&message.fee.native_fee.0.to_le_bytes());
+        data.extend_from_slice(message.sender.to_string().as_bytes());
+        data.extend_from_slice(message.msg.as_bytes());
+        data.extend_from_slice(signer_id.as_bytes());
```

Comment:
I think we can use borsh encoding instead.
Idk why we need the signer here, but it also can be encoded using borsh